components/header/wallet_modal.tsx

```diff
@@ -0,0 +1,125 @@
+import { Dispatch, FormEvent, SetStateAction, useState } from 'react';
+import Image from 'next/image';
+import ReactModal from 'react-modal';
+
+import { validateWalletAddress } from '@utils';
+
+import arconnectLogo from '../../public/arconnect.png';
+import {
+	CloseModalButton,
+	ImportWalletErrorMessage,
+	ImportWalletForm,
+	ImportWalletInput,
+	ModalButton,
+	ModalButtons,
+	ModalContent,
+	ModalText,
+} from './wallet_modal.style';
+
+const WalletModal = ({
+	isOpen,
+	closeModal,
+	setWalletAddress,
+}: {
+	isOpen: boolean;
+	closeModal: Function;
+	setWalletAddress: Dispatch<SetStateAction<string>>;
+}): JSX.Element => {
+	const [isArconnectAvailable, setIsArconnectAvailable] = useState(false);
+	const [showWalletInput, setShowWalletInput] = useState(false);
+	const [importedPublicWalletAddress, setImportedPublicWalletAddress] = useState('');
+	const [importedPublicWalletAddressError, setImportedPublicWalletAddressError] = useState('');
```

Comment:
Any time we're tracking this much state in a single React component I worry about the complexity multiplying as this component might grow and how much prop drilling it will take to distribute this state to child components

We can always consider a custom hook to encapsulate the state with a tidy API for the component to consume. But I don't think that would be worth it here, as this project is well-scoped and looking great.

Reply:
I also agree with that. For now, we can keep like it is. Let's review this component next week.